tests/v2/validation/fleet/public_repo_test.go

```diff
@@ -0,0 +1,111 @@
+//go:build validation || airgap
+
+package fleet
+
+import (
+	"testing"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/v2/actions/provisioninginput"
+	"github.com/rancher/shepherd/clients/rancher"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	extensionscluster "github.com/rancher/shepherd/extensions/clusters"
+	extensionsfleet "github.com/rancher/shepherd/extensions/fleet"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FleetPublicRepoTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	provisioningConfig *provisioninginput.Config
+}
+
+func (f *FleetPublicRepoTestSuite) TearDownSuite() {
+	f.session.Cleanup()
+}
+
+func (f *FleetPublicRepoTestSuite) SetupSuite() {
+	f.session = session.NewSession()
+
+	client, err := rancher.NewClient("", f.session)
+	require.NoError(f.T(), err)
+
+	f.client = client
+
+	userConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, userConfig)
+
+	f.provisioningConfig = userConfig
+}
+
+func (f *FleetPublicRepoTestSuite) TestGitRepoDeployment() {
+	defer f.session.Cleanup()
+
+	var clusterID string
+	var err error
+	clusterObject, _, _ := extensionscluster.GetProvisioningClusterByName(f.client, f.client.RancherConfig.ClusterName, "fleet-default")
+	if clusterObject != nil {
+		status := &provv1.ClusterStatus{}
+		err := steveV1.ConvertToK8sType(clusterObject.Status, status)
+		require.NoError(f.T(), err)
+
+		clusterID = status.ClusterName
+		logrus.Info(clusterID)
+	} else {
+		clusterID, err = extensionscluster.GetClusterIDByName(f.client, f.client.RancherConfig.ClusterName)
+		require.NoError(f.T(), err)
+	}
+
+	podErrors := pods.StatusPods(f.client, clusterID)
+	require.Empty(f.T(), podErrors)
+
+	fleetVersion, err := getDeploymentVersion(f.client, fleetControllerName, "local")
```

Comment:
how do I know what the expected version is programmatically?

Reply:
thinking about this more, I would say the only way to know ahead of time is to add a spot for it in the config. but IMO this isn't necessary; as long as the version is reported in Qase / the test, then we can directly see what version we're validation on. I want to avoid adding stuff to the config, and in this case I don't think its needed. But I can see a case for it if others want this added in.

Reply:
something like this would work here: https://github.com/rancher/rancher/blob/main/tests/v2/validation/charts/webhook_test.go#L49

Reply:
hmm. maybe we can get on a call for this, but currently I don't think its necessary to check the version.